notification/tasks.py

```diff
@@ -0,0 +1,196 @@
+import smtplib
+from collections import defaultdict
+from datetime import datetime
+
+from celery import shared_task
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from notification.email_sender import send_the_email
+from notification.models import Notification
+from celery.utils.log import get_task_logger
+
+logger = get_task_logger(__name__)
+
+
+def report_notifications_upcoming_events_errors_for_datasteward(user):
+    """
+    Send upcoming events notifications errors report for datasteward, if any.
+
+    Params:
+        user: The user whom the notifications sending failed
+    """
+    logger.info("Sending upcoming events notifications errors for datasteward")
+
+    notifications_not_processed = Notification.objects.filter(
+        recipient=user.id, dispatch_by_email=True, processing_date=None
+    )
+
+    # Send email to data steward in case of errors
+    if notifications_not_processed:
+        logger.error("Some notification are not processed: ")
+        logger.error(notifications_not_processed)
+
+        notifications_not_processed_by_content_type = defaultdict(list)
+
+        for notif in notifications_not_processed:
+            notifications_not_processed_by_content_type[notif.content_type].append(
+                notif
+            )
+
+        context = {
+            "notifications": dict(notifications_not_processed_by_content_type),
+            "error_message": "Please find below the notifications that failed to be sent to user: "
+            + user.full_name,
+        }
+        try:
+            send_the_email(
+                settings.EMAIL_DONOTREPLY,
+                settings.DATASTEWARD_MAILING_LIST,
+                "Notifications",
+                "notification/email_datasteward_notifications_error",
+                context,
+            )
+        except Exception as e:
+            logger.error(
+                f"Failed: An error occurred while sending Email notification error report for data-stewards."
+                f" Error: {e}"
+            )
+            print(
+                f"Failed: An error occurred while sending Email notification error report for data-stewards."
+                f" Error: {e}"
+            )
```

Comment:
This could be removed no? Also, is it intentional for the error to not be raised again?

Reply:
yes it could be removed. The idea was that we log the error and then send datastewards an email in case of notification error and continue to the next user. you think we should raise the error anyways?

Reply:
It makes sense. I think the error email should be sent at the very end though, or we risk sending an email for every user on some days.

Reply:
i thought it is one email per user. I felt it would be easier for data stewards to track which user notifications errors they checked and which they didn't. Otherwise. if all users in one email it might be long and maybe they will not be able to keep track of where they stopped.

Reply:
sorry, removed it now.